### PR TITLE
add integration test suite

### PR DESCRIPTION
**Pull request by adithyasolai**
Leveraged TestContainers framework to create integration tests that spin up a small MySQL DB in a local Docker environment and test MvcController functionality.

Added the following test cases:
- Simple Deposit (customer account not in overdraft or frozen, etc)
- Simple Withdraw (withdraw does not cause overdraft, customer account not frozen, etc)
- Withdraw that causes customer to go into Overdraft
- Deposit that repays all of the overdraft balance AND adds excess deposit amount to main balance
- Deposit that repays only some of the overdraft balance
- Fraud dispute that reverses a simple deposit transaction
- Fraud dispute that reverses a simple withdraw transaction

**Changes Made in Each File**
___
`.gitignore`
- Running Jython scripts creates a Jython cache in local dev environment that should be gitignored. No Jython in this PR, but likely will be added later to use the existing python scripts that populate the MySQL DB (like `addCustomers.py`).
___
`pom.xml`
- Changed versions of existing dependencies like `spring-boot-starter-parent` and `junit` to be compatible with the new testing frameworks added.
- Added new dependencies to support integ testing like test-containers, hamcrest, jython, and surefire.
___
`src/main/java/net/codejava/MvcController.java`
- Standardized names of MySQL DB tables in SQL queries to be compatible with all versions of MySQL. The MySQL version used in Integ Tests is different from the MySQL DB version used by the actual application.
- Fixed a bug that previously did not increment `numFraudReversals` on deposits. Kind of a scope creep, but it was picked up by one of the integ tests in this PR.
- Made some `private static` fields like `INTEREST_RATE` public so that they can re-used in the Integ Tests.
___
`src/test/java/net/codejava/MvcControllerIntegTest.java`
- Contains the integration tests for the 7 test cases described above.
- Also includes helper methods that facilitate the integration tests and help make the code more readable.
___
`src/test/java/net/codejava/MvcControllerTest.java`
- Removed existing unit tests that cover the same cases covered by the new Integ tests.
- Standardized MySQL Table names in SQL queries here as well (part of the stubbing of Controller SQL queries)
___
`src/test/resources/createDB.sql`
- Script run only once `@BeforeAll` Integ tests that creates empty tables to match the application's desired DB schema.
___
`src/test/resources/clearDB.sql`
- Script run after every Integ Test that wipes all rows in all tables in the MySQL DB.
___
`src/test/resources/logback-test.xml`
- New Config file in test directory to reduce the logging produced by test containers and speed up integ tests.
___
**Future Changes to Make**
- Research how to `pip install` packages like `pymysql` needed for existing MySQL DB-populating scripts like `addCustomers.py` so that they can be run before Integ Tests using Jython.
- Move all Integ Test helper methods to another class in the `tst` directory, and make all those methods `public static` instead of `private static`.
- Write Integ Tests for Dispute Reversal edge cases that involve a customer being in Overdraft.
- Write Integ Test for Overdraft edge case where customer is already in overdraft and tries to withdraw an amount that exceeds Overdraft limit.

### CODE REVIEW

File: src/test/resources/hello.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print("Hello, world!")
```
Comment (s1ddh4rthc) [inline src/test/resources/hello.py]: Is this a necessary line to keep?

Comment (adithyasolai) [reply src/test/resources/hello.py]: Was deleted in a future commit. Don't read this PR commit-by-commit. Only look at the final changes in the "Files Changed" tab above.
